usr/sbin/regionsrv-enabler-azure

```diff
@@ -12,6 +12,40 @@
 # Lesser General Public License for more details.
 # You should have received a copy of the GNU Lesser General Public
 # License along with this library.
+
+"""
+Check the different states of the license type:
+START -> The state of the instance when this code is executed
+         This code is executed by a timer
+
+    START   LICENSE CHANGED TO     CACHE   ACTION
+1   BYOS-1  PAYG                   No      Register - Update Infra
+2   BYOS-1  PAYG                   BYOS    Register - Update Infra
+3   BYOS-1  BYOS-1                 No      Do nothing
+4   BYOS-1  BYOS-1                 BYOS    Do nothing
+5   BYOS-1  BYOS-2                 No      Do nothing
+6   BYOS-1  BYOS-2                 BYOS    Do nothing
+
+7   BYOS-2  PAYG                   No      Do nothing
```

Comment:
If the instance is switched to PAYG and the cache is not PAYG we always have to re-register.